core/livepeernode.go

```diff
@@ -63,6 +63,13 @@ const (
 	Bootnode
 )
 
+type SegmentChan chan *SegChanData
+
+type SegChanData struct {
+	seg *SignedSegment
+	res chan error
```

Comment:
It's actually okay/expected to return a non-error (nil) value here since that means the transcoding succeeded. Still, I can make the change if you think it helps the clarity.

If Go was a language with proper sum types, `res` would be a `Maybe error` rather than a bare `error` type.

Reply:
Ah I see - I wasn't taking into account the usage of the `res` channel to block until the worker sends a signal that processing of a particular segment is complete. With that in mind I think keeping the name makes sense.